code/modules/power/turbine/turbine_computer.dm

```diff
@@ -54,20 +54,20 @@
 	var/list/data = list()
 
 	var/obj/machinery/power/turbine/core_rotor/main_control = turbine_core?.resolve()
-
-	data["connected"] = main_control ? TRUE : FALSE
+	data["connected"] =  QDELETED(main_control) ? TRUE : FALSE
 	if(!main_control)
 		return
+
 	data["active"] = main_control.active
 	data["rpm"] = main_control.rpm ? main_control.rpm : 0
 	data["power"] = main_control.produced_energy ? main_control.produced_energy : 0
-	data["temp"] = main_control.compressor.input_turf?.air.temperature
 	data["integrity"] = main_control.get_turbine_integrity()
 	data["parts_linked"] = main_control.all_parts_connected
 	data["parts_ready"] = main_control.all_parts_ready()
 
 	data["max_rpm"] = main_control.max_allowed_rpm
 	data["max_temperature"] = main_control.max_allowed_temperature
+	data["temp"] = QDELETED(main_control.compressor) ? 0 : main_control.compressor.input_turf?.air.temperature
```

Comment:
```suggestion
	data["temp"] = main_control.compressor?.input_turf?.air.temperature || 0
```
Provided the `compressor` variable of `main_control` gets properly cleared up on deletion as it should anyway, this is fine.